packages/common-library/src/common_library/async_tools.py

```diff
@@ -89,32 +89,37 @@ async def cancel_wait_task(
         TimeoutError: raised if cannot cancel the task.
         CancelledError: raised ONLY if owner is being cancelled.
     """
+    if task.done():
+        # nothing to do here
+        return
 
-    cancelling = task.cancel()
-    if not cancelling:
-        return  # task was alredy cancelled
-
-    assert task.cancelling()  # nosec
-    assert not task.cancelled()  # nosec
-
+    # mark for cancellation
+    task.cancel("cancel_wait_task was called to cancel this task")
     try:
-
+        _logger.debug("Cancelling task %s", task.get_name())
         await asyncio.shield(
             # NOTE shield ensures that cancellation of the caller function won't stop you
             # from observing the cancellation/finalization of task.
             asyncio.wait_for(task, timeout=max_delay)
         )
-
-    except asyncio.CancelledError:
-        if not task.cancelled():
-            # task owner function is being cancelled -> propagate cancellation
-            raise
-
-        # else: task cancellation is complete, we can safely ignore it
-        _logger.debug(
-            "Task %s cancellation is complete",
+    except TimeoutError:
+        _logger.exception(
+            "Timeout while cancelling task %s after %s seconds",
```

Comment:
THOUGHT: instead of logging & re-raise, why not create a new exception inherited from TimeoutError that contains this meesage, e.g.

```python
class TaskCancelTimeout(TimeoutError):
    pass

...

except TimeoutError as e:
    raise TaskCancelTimeout(
        f"Timeout while cancelling task {task.get_name()} after {max_delay} seconds"
    ) from e

```

This approach
- avoids re-logging in case the TimeoutError is logged again on the handling side
- lets callers explicitly catch TaskCanceTimeout if they need special handling
- Still behaves like a TimeoutERror for generic catches
- Message and context preseved

Reply:
OK, but just to be clear: it is not about the amount of error messages but the repetition of them ... (I will show you some examples of how confusing is to see tracebacks of the same errors multiple times)  :-)